### DISCUSSION TITLE
iceoryx v3.0 release coordination

### DISCUSSION THREAD

**Discussion by elBoberido** [eclipse-iceoryx/iceoryx#2189] [General]
@FerdinandSpitzschnueffler @mossmaurice @dkroenke @MatthiasKillat @budrus @elfenpiff  please have a look at https://github.com/eclipse-iceoryx/iceoryx/projects/5 and check if there is something in the `To do` column which you want to have for the v3.0 release. Except `Release validation` which should be reduced to the tasks we did for v1.0, I think everything can easily be postponed to v4.0 since contain bugs that are either fixed and just need some confirmation or issues which are also present in v2.0.

Some of the items in the `In progress` column are also no blocker for a release and the others are basically done and just need a final entry in the release notes.

**Comment by mossmaurice** [comment]
@elBoberido 

> check if there is something in the `To do` column which you want to have for the v3.0 release.

I went over all the issues in the project under "To do" and "progress".

My recommendation and effort estimates:

## Must-haves

- Find a name for the release by voting as we did it for the previous releases :coffee: :carrot: :cake: :chocolate_bar: 
- #1295, for me the docs are definitely a must-have as otherwise it will be confusing for users and there will be reports like #2134 
   - ~5 days effort, as the build of the docs is currently broken due to the change to relative links
- #1541, could be addressed together with the release validation 
   - ~3h effort
- #1616, sounds like we should address this topic before the `v3.0` release 
   - ~6h efforts, but rather hard to estimate for me, please double check :construction: 

## Can be postponed to `v4.0`

- #590 :heavy_check_mark: 
   - Nearly done, a few `@todo`s still in the code
-  #1391, not a blocker for the release, just a nice to have :heavy_check_mark: 
- #1921, not a blocker for the release, just a nice to have :heavy_check_mark: 
- #1593 was already done :heavy_check_mark: 
- #1432, documentation can be done for the next release :heavy_check_mark: 
- #1032, the scope of this issue is a bit unclear to me, I think it can be postponed, please double-check :construction: 
- #1293, just needs some documentation of the approach e.g. in `CONTRIBUTING.md`, please double-check :construction: 
- #1646 I don't think this is actually an issue, please double check :construction: 

> Except `Release validation` which should be reduced to the tasks we did for v1.0, I think everything can easily be postponed to v4.0 since contain bugs that are either fixed and just need some confirmation or issues which are also present in v2.0.


I agree, besides finding the release name and basic release validation of the docs, most of the issues can be postponed to `v4.0`

**Reply by elBoberido** [reply]
- #1032: This is actually done but the issue description does not match the actual work and needs to be updated
- #1646: From my point of view this is not a problem. I already looked at it some time ago but I need confirmation from a second developer. This is a non-blocker from my point of view.
- #1293: We can keep this in v3.0 but handle it as non-blocker
- #1541: This is s a documentation issue and not an actual bug but it should be fixed for v3.0
- #1616: Indeed, it should not be too hard to fix
- #1295: This is something we need to cut down to the minimum else the release will be dragged out for months
- #2202: This came in recently and should be fixed for v3.0; either a proper fix or at least the documentation needs to state the issue and how to prevent it
- #1293: Yes, needs some documentation but also not a blocker.
- #590: Yup, can be postponed
- #1921: This can also be postponed
